ruby/lib/minitest/queue/runner.rb

```diff
@@ -37,15 +37,22 @@ def run!
       end
 
       def retry_command
-        reset_counters
-        retry_queue = queue.retry_queue
-        unless retry_queue.exhausted?
-          self.queue = retry_queue
-        end
-        run_command
+        STDERR.puts "Warning: the retry subcommand is deprecated."
+        run_command # aliased for backward compatibility purpose
       end
 
       def run_command
+        if queue.retrying?
+          reset_counters
+          retry_queue = queue.retry_queue
+          if retry_queue.exhausted?
+            puts "The retry queue does not contain any failure, we'll process the main queue instead."
```

Comment:
I figured some output on the selected behavior would help future debugging. Feel free to suggest alternative phrasing as english is not my strong suit.

Reply:
Some possible alternatives:

> The retry queue does not contain any failures, we'll process the main queue instead.

> Found 0 tests to retry, processing the main queue.

> This worker has no previous test failures, so nothing to retry. Processing the main queue.
>